runtime/rasdump/javadump.cpp

```diff
@@ -5747,8 +5888,18 @@ UDATA
 protectedGetVMThreadName(struct J9PortLibrary *portLibrary, void *args)
 {
 	void **parameters = (void**)args;
-
-	return (UDATA)tryGetOMRVMThreadName(((J9VMThread*)parameters[1])->omrVMThread);
+#if JAVA_SPEC_VERSION >= 21
+	J9VMThread *vmThread = (J9VMThread *)parameters[1];
+	if ((NULL != vmThread->threadObject) && IS_JAVA_LANG_VIRTUALTHREAD(vmThread, vmThread->threadObject)) {
+		/* For VirtualThread, get name from threadObject directly. */
+		j9object_t nameObject = J9VMJAVALANGTHREAD_NAME(vmThread, vmThread->threadObject);
+		const char *threadName = getVMThreadNameFromString(vmThread, nameObject);
+		BOOLEAN *isVThread = (BOOLEAN *)parameters[2];
+		*isVThread = TRUE;
+		return (UDATA)threadName;
+	}
+#endif /* JAVA_SPEC_VERSION >= 21 */
+	return (UDATA)tryGetOMRVMThreadName(((J9VMThread *)parameters[1])->omrVMThread);
```

Comment:
Suggest extracting `vmThread` for reuse on line 5902:
```c
	void **parameters = (void **)args;
	J9VMThread *vmThread = (J9VMThread *)parameters[0];
#if JAVA_SPEC_VERSION >= 21
	if ((NULL != vmThread->threadObject) && IS_JAVA_LANG_VIRTUALTHREAD(vmThread, vmThread->threadObject)) {
		/* For VirtualThread, get name from threadObject directly. */
		j9object_t nameObject = J9VMJAVALANGTHREAD_NAME(vmThread, vmThread->threadObject);
		char *threadName = getVMThreadNameFromString(vmThread, nameObject);
		BOOLEAN *isVThread = (BOOLEAN *)parameters[1];
		*isVThread = TRUE;
		return (UDATA)threadName;
	}
#endif /* JAVA_SPEC_VERSION >= 21 */
	return (UDATA)tryGetOMRVMThreadName(vmThread->omrVMThread);
```

Reply:
Declaring `const char *threadName` is misleading; please remove `const` because in the case of a virtual thread it must be freed.

Reply:
fixed